pool_dynamics_test.go

```diff
@@ -15,8 +17,9 @@ import (
 )
 
 const (
-	nodesSize = 200
+	nodesSize = 6
 )
```

Comment:
Why do we need to do this ?

Reply:
This is used to change the pool size desired for testing. I reduced the pool size for testing cache affinity to make it easier to debug and get the tests passing. Once we get that passing we can set it to a high / realistic number again.

Reply:
Okay, yeah that makes sense.

Reply:
This has already been structured so that the lower pool target is only used in the test context. A larger pool size target is used when running normally